src/XmlAgent.cc

```diff
@@ -850,21 +850,15 @@ YCPValue XmlAgent::Execute(const YCPPath &path, const YCPValue& value, const YCP
 
     YCPValue result = YCPVoid();
     int size;
-    const char * input = "";
-    const char *content = "";
+    std::string pth;
```

Comment:
when I see all this logic, maybe even more straightforward way would be to define here `bool string_output = false` and below when you assign "string" to pth simple set `string_output = true`. What do you think?

Reply:
That makes sense